gsl/gsl_byte

```diff
@@ -106,6 +106,12 @@ constexpr IntegerType to_integer(byte b) noexcept
     return {b};
 }
 
+constexpr byte to_byte(unsigned char i) noexcept
+{
```

Comment:
This version allows "accidental" narrowing conversion. Could you cook a version that weed out non-'unsigned char' via sfinae trickery?  E.g. something along the line of
```
template<typename T, typename = std::enable_if_t<std::is_same<T, unsigned char>::value, T>>
constexpr byte to_byte(T t) noexcept
{
     return byte(t);
}
```